docs/src/languages/nodejs/debug.md

```diff
@@ -3,121 +3,244 @@ title: "Debugging"
 
-```sh
-$ export PLATFORM_RELATIONSHIPS="$(platform tunnel:info --encode)"
-$ export PORT=8888
-```
+{{< codetabs >}}
 
-### Run your application in inspect mode
+---
+title=Using Chrome developer tools
+file=none
+highlight=false
+---
 
-```sh
-$ node --inspect node --inspect <js_file>
-```
+Go to `chrome://inspect`.
+Find your running app under the `Remote Target` list.
+Click **inspect** to start the debugger.
+
+<--->
+
+---
+title=Using Visual Studio Code
+file=none
+highlight=false
+---
+
+Use the `Node.js: Attach` debugger option.
+
+If you haven't created the option:
+
+1. On the **Run and Debug** tab, click `create a launch.json file`.
+2. Select `Node.js` as the environment.
+3. In the `configurations` array, start IntelliSense (usually <kbd>ctrl</kbd>+<kbd>space</kbd>).
+4. Select `Node.js: Attach`.
+5. Make sure the port is the same as in [step 4 above](#4-if-debugging-remotely-forward-the-debugger-port-locally).
+
+Once you have the option:
+
+In the **Run and Debug** tab, select `Attach` from the menu and click **Start Debugging** (the green arrow).
+
+<!--vale off -->
+{{< /codetabs >}}
+<!--vale on -->
+
+Now when you load the site at your deployed URL (if debugging remote) or localhost (if debugging locally),
+the local debugger you've attached is called.
+
+Set breakpoints:
+
+{{< codetabs >}}
+
+---
+title=Remote
+file=none
+highlight=false
+---
+
+In the JavaScript files from your remote site:
+On the **Run and Debug** tab under **Loaded Scripts** find `Attach: Remote Process` > `/app`.
+
+<--->
+
+---
+title=Local
+file=none
+highlight=false
+---
 
-`js_file` being your application entrypoint.
+Directly in your source files.
 
-### Connect the debugger
+<!--vale off -->
+{{< /codetabs >}}
+<!--vale on -->
 
-You can now connect your debugger just like you would with any locally running node process.
 
-#### Using the chrome devtools
+## Other issues
 
-Go to `chrome://inspect`. You should see the running application under the `Remote Target` list and be able to click the `inspect` link to open the debugger.
+### pm2 process manager blocks other processes
 
-#### Using VSCode
+If you're using the [`pm2` process manager](https://github.com/unitech/pm2) to start your app from a script,
+you might find it daemonizes itself and blocks other processes (such as backups) by constantly respawning.
+This may happen even if you use the `--no-daemon` flag.
 
-Use the `Node.js: Attach to process` debugger option. If not configured already, go to the debugger tab and select `create a launch.json file` and select `Node.js`. Then use intellisense (usually `control`+`space`) in the `configurations` array. Select `Node.js: Attach to process`. You can now select `Attach by process ID` in the `RUN AND DEBUG` drop-down and click the green arrow. You can now set breakpoints in your source files directly.
+Instead of using a script, call `pm2 start` directly in your [`start` command](./_index.md#4-start-your-app).
```

Comment:
:question:

Shouldn't it [be `pm2 start app.js`](https://pm2.keymetrics.io/docs/usage/quick-start/) if the app's entry point is `app.js`?

```suggestion
Instead of using a script, call `pm2 start <app name>` directly in your [`start` command](./_index.md#4-start-your-app).
```

Reply:
Technically the entire command should be something like:
```bash
PM2_HOME=/app/run pm2 start <START_FILE> --no-daemon
```
But given that we're no longer suggesting `pm2` as the primary way to go about Node.js apps, it didn't seem worth it to go into the details. They will only see this error if they have the complete command already in a script. So it's just about moving whatever command they have into the start command.